src/main/java/org/alliancegenome/curation_api/dao/GeneDAO.java

```diff
@@ -19,10 +20,46 @@ public Gene getByIdOrCurie(String id) {
         return find(id);
     }
 
-    public List<String> findAllAnnotationIds(String taxonID) {
-        Query jpqlQuery = entityManager.createQuery("SELECT annotation.curie FROM Gene annotation WHERE annotation.taxon.curie=:taxonId");
-        jpqlQuery.setParameter("taxonId", taxonID);
+    public List<String> findAllCuriesByTaxon(String taxonId) {
+        Query jpqlQuery = entityManager.createQuery("SELECT gene.curie FROM Gene gene WHERE gene.taxon.curie=:taxonId");
+        jpqlQuery.setParameter("taxonId", taxonId);
         return (List<String>) jpqlQuery.getResultList();
     }
+    
+    @Transactional
+    public void deleteReferencingDiseaseAnnotations(List<String> geneCuries) {
```

Comment:
There may be a better method than this...

Reply:
Probably, but I couldn't find a better way of cascading deletes where @ManyToMany relationships are involved

Reply:
Deleting the object should remove all the entries in each of those tables. If not then we need to look at the correct combination of CASCADE DELETE etc. And another option is to Pull the object from the database clear all the lists save it then delete it.

Reply:
I tried various combinations of cascade options but was getting nowhere.  I was trying to follow the logic described here: [https://thorben-janssen.com/avoid-cascadetype-delete-many-assocations/](https://thorben-janssen.com/avoid-cascadetype-delete-many-assocations/)